lib/grandpa/message_handler.go

```diff
@@ -561,6 +561,10 @@ func (s *Service) VerifyBlockJustification(hash common.Hash, justification []byt
 		return err
 	}
 
+	if !hash.Equal(fj.Commit.Hash) {
+		return fmt.Errorf("%w: expected %s. got %s", ErrJustificationMismatch, fj.Commit.Hash.Short(), hash.Short())
```

Comment:
This will generate too much noise in the error message:
`justification does not correspond to given block hash: justification commit hash is %s and does not match hash %s`

Another way could be:
`justification does not correspond to given block hash: justification %s. block hash %s`

Reply:
Yeah sounds good 👍 just that full stop `.` is weird to me, maybe just replace it with `and` 😉